Pass `XP_*` environment variables

thekid opened the pull request:
## What's in it

Contains exe, version, execution model and subcommand as suggested in https://github.com/xp-runners/reference/issues/28#issuecomment-224088746. 
### Example

``` php
[
  XP_EXE => "C:\tools\cygwin\home\Timm\devel\runners\reference\xp.exe"
  XP_VERSION => "7.5.1.17"
  XP_MODEL => "default"
  XP_COMMAND => "write"
]
```
## Usecases

The path to the binary can be used to calculate path to software and config files delivered alongside the XP runners, e.g. certificate bundle like suggested in xp-framework/core#150

thekid committed:
Pass `XP_CMD` environment variable containing exe, version, execution model and subcommand

See https://github.com/xp-runners/reference/issues/28#issuecomment-224088746

thekid commented:
Args, Mono doesn't have [StartInfo.Environment](https://msdn.microsoft.com/en-us/library/system.diagnostics.processstartinfo.environment%28v=vs.110%29.aspx):

_src/xp.runner/Command.cs(117,28): error CS1061: Type `System.Diagnostics.ProcessStartInfo` does not contain a definition for `Environment` and no extension method `Environment` of type `System.Diagnostics.ProcessStartInfo` could be found. Are you missing an assembly reference?_ 

This is a .NET 4.6 feature

thekid committed:
Rewrite proc.StartInfo.Environment.Add() -> Environment.SetEnvironmentVariable()

See https://travis-ci.org/xp-runners/reference/builds/135718495

thekid committed:
Use StartInfo.EnvironmentVariables instead of .Environment

See https://github.com/dotnet/corefx/issues/315

thekid commented:
I don't like the name XP_CMD - maybe XP_ENVIRONMENT?

kiesel commented:
Why not pass each data item as separate environment variable, each prefixes by `XP_RUNNER_`? Having them all crammed in one variable means you have to parse the line and extract the information you need from there.

thekid commented:
> Why not pass each data item as separate environment variable, each prefixes by  XP_RUNNER_ ? 

Hrm, seems like a bit more overhead in the calling side... Lots of environment add operations. On the other hand it's more efficient in the called side (PHP).

I'll look into how that would affect the code though.

thekid committed:
Use separate environment variables

- XP_EXE: Path to XP runner binary (e.g. /usr/bin/xp)
- XP_VERSION: Version (e.g. 7.5.1.17)
- XP_MODEL: Execution model (e.g. default, watch or supervise)
- XP_COMMAND: Subcommand name used

thekid commented:
> I'll look into how that would affect the code though.

I've decided to go the way you suggested, this makes PHP code easier, and only adds a negligible overhead to C#.
